docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,326 @@
+javac -processor resourceleak MyFile.java ...
+javac -processor org.checkerframework.checker.resourceleak.ResourceLeakChecker MyFile.java ...
+\end{Verbatim}
+
+The Resource Leak Checker supports all the command-line arguments
+ listed in Section~\ref{called-methods-run-checker} for
+the Called Methods Checker.
+
+\sectionAndLabel{Resource Leak Checker annotations}{resource-leak-annotations}
+
+The Resource Leak Checker relies on the type qualifiers of two other checkers:
+the Must Call Checker (Section~\ref{must-call-annotations}) and
+the Called Methods Checker (Section~\ref{called-methods-spec}). You might need
+to write qualifiers from either type hierarchy. The most common annotations from
+these checkers that you might need to write are:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{MustCall}{String[] value}]
+for example on an element with compile-time type \<Object> that might contain a \<Socket>.
+See Section~\ref{must-call-annotations}.
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{InheritableMustCall}{String[] value}]
+on any classes defined in your program that have must-call obligations. See Section~\ref{must-call-on-class}.
+
+\item[\refqualclass{checker/calledmethods/qual}{EnsuresCalledMethods}] on a method in
+your code that fulfills a must-call obligation of one of its parameters or of a field.
+See Section~\ref{called-methods-ensurescalledmethods}.
+
+\end{description}
+
+The Resource Leak Checker supports annotations that express
+aliasing patterns related to resource leaks:
+
+\begin{description}
+
+\item[\refqualclass{checker/mustcall/qual}{Owning}]
+\item[\refqualclass{checker/mustcall/qual}{NotOwning}]
+  expresses ownership.  When two aliases exist to the same Java object,
+  \<@Owning> and \<@NotOwning> indicate which of the two is responsible for
+  fulfilling must-call obligations.  For details, see
+  Section~\ref{resource-leak-ownership}.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallAlias}]
+  represents a ``resource-aliasing'' relationship.  Resource aliases are
+  distinct Java objects that control the same resource(s):
+  fulfilling the must-call obligations of one object also
+  fulfills the obligations of the other object.  For details,
+  see Section~\ref{resource-leak-resource-alias}.
+
+\end{description}
+
+The Resource Leak Checker also supports an annotation to permit re-assigning
+fields or re-opening resources:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{CreatesMustCallFor}{String value}]
+  is a declaration annotation that indicates that the annotated method ``resets'' the must-call
+  obligations of some expression.  Multiple \<@CreatesMustCallFor>
+  annotations can be written on the same method.  For more details
+  on obligation creation, see Section~\ref{resource-leak-createsmustcallfor}.
+
+\end{description}
+
+
+\sectionAndLabel{Aliased references and ownership transfer}{resource-leak-ownership}
```

Comment:
I think a general example would be helpful before jumping into the more advance features.

Reply:
I wrote one up and added it to the manual as its own section. I'm a little worried I may have gone overboard on details, so if you could take a look and let me know what you think I'd appreciate it.

Reply:
It's a little much, but I would leave it.  I think it makes the other sections easier to understand.